### PR TITLE
Add authentication and S2IT's email validation

### PR DESCRIPTION
**Pull request by mdsrosa**
I added the authentication controller in order to prevent unauthenticated users to access the main page.
I also added an e-mail validation, so that only emails with the domain of S2IT (s2it.com.br) can register. I think in the future we can improve it by making a check of user existence in the domain of S2IT.

I am open to any change that is needed in this PR.

### CODE REVIEW

File: app/validators/email_validator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    begin
+      m = Mail::Address.new(value)
+      r = m.domain != nil && m.domain.match('s2it\.com\.br') && m.address == value
```
Comment (fpgentil) [inline app/validators/email_validator.rb]: Wouldn't be better to leave this `'s2it\.com\.br'` inside an ENV variable?

File: app/validators/email_validator.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    rescue Exception => e
+      r = false
+    end
+    record.errors[attribute] << (options[:message] || "is not a S2IT email") unless r
```
Comment (fpgentil) [inline app/validators/email_validator.rb]: Also here, personally I don't like these `hard-coded` messages.. :smile:

File: app/validators/email_validator.rb (hunk 1 above)
Comment (mdsrosa) [inline app/validators/email_validator.rb]: Yes. I agree. I didn't know to make "ENV variables" at the time. Now I do. :smile:

File: app/validators/email_validator.rb (hunk 2 above)
Comment (mdsrosa) [inline app/validators/email_validator.rb]: Yeah. I don't like it either. I'm going to add the ENV variable then I'll change this to a more generic message. :smile:

**Comment by mdsrosa** [comment]
I added the VALID_EMAIL_DOMAIN_REGEX and changed the validation error message.

**Comment by duduribeiro** [comment]
You can setup the devise to connect on the LDAP server too.. S2IT has one..

File: config/application.rb [hunk 3]
```diff
@@ -31,5 +31,13 @@ class Application < Rails::Application
...
     # Do not swallow errors in after_commit/after_rollback callbacks.
     config.active_record.raise_in_transactional_callbacks = true
+
+    # load all env variables
+    config.before_configuration do
+      env_file = File.join(Rails.root, 'config', '.env.yml')
+      YAML.load(File.open(env_file)).each do |key, value|
+        ENV[key.to_s] = value
+      end if File.exists?(env_file)
+    end
```
Comment (duduribeiro) [inline config/application.rb]: https://github.com/bkeepers/dotenv :P

**Comment by mdsrosa** [comment]
@duduribeiro Thank you for your insights! I'll consider this when I get back to work on this project. But, in the meantime, feel free to contribute. :smile:

**Pull request merged**